frontend/src/components/article-table/article-table.tsx

```diff
@@ -96,16 +96,42 @@ export default function ArticleTable({ data }: { data: Article[] }) {
           <h2 className=" text-2xl text-black mb-4 w-full text-start border-b border-black pb-2">
             Articles
           </h2>
-          <div className="flex flex-row justify-between gap-4">
+          <div className="flex flex-col gap-4">
             <Input
-              placeholder="Search title..."
-              value={table.getColumn('author')?.getFilterValue() as string}
-              onChange={(event) =>
-                table.getColumn('author')?.setFilterValue(event.target.value)
+              ref={searchInputRef}
+              placeholder="Search..."
+              value={
+                table.getColumn(selectedFilter)?.getFilterValue() as string
               }
-              className="max-w-xs"
+              onChange={(event) => {
+                setColumnFilters([]);
+                table
+                  .getColumn(selectedFilter)
+                  ?.setFilterValue(event.target.value);
+              }}
             />
-            <Button>Search</Button>
+            <RadioGroup
+              defaultValue="Title"
+              className="flex flex-row"
+              onValueChange={(value) => {
```

Comment:
don't think it affects anything, but the filter value should have a type

Reply:
nvm, dependency install fixed the error. there is an unused variable on line 32 
`
  const [sorting, setSorting] = useState<SortingState>([]);`

But other than that lgtm